### PR TITLE
Improve Project Workflows

### LINKED ISSUE #41: Improve project workflows
# Issue | Checkers

## 1. Type of Issue

- [ ] Bug  
- [ ] Feature Request  
- [x] Enhancement  
- [ ] Documentation  
- [ ] Other  

---

## 2. Brief Description  

Current workflows tests the dojo contracts,but need to improve the way of installation. At the same time we are not checking the code format

## 3. Current Behavior  
  
1. Current version of dojo is hardcoded
2. Missing scarb fmt or any linter for dojo/cairo code
---

## 4. Expected Behavior  
  
 1. it should install the dojo version from .tool-versions.toml file
 2. Implent the script to check dojo code format
---
---

## 4. Proposed Solution (Optional)  

Take as reference this workflows: 
- [Eternum](https://github.com/BibliothecaDAO/eternum/tree/next/.github) 
- [zKube](https://github.com/z-korp/zkube/blob/main/.github/workflows/ci.yaml)

### PR DESCRIPTION
**Pull request by No-bodyq**
# Pull Request | Checkers

## 1. Issue Link

- Closes #41 

## 2. Brief Description

Install dojo dynamically from .tools-versions file and add scarb fmt check

## 3. Changes Made

- [x] it should install the dojo version from .tool-versions.toml file
- [x] Implent the script to check dojo code format

## 4. Checklist

- [x] Tests added/updated
- [ ] Documentation updated (if needed)
- [ ] Code follows standards

### CODE REVIEW

**Comment by No-bodyq** [comment]
Please review @gianalarcon

**Comment by No-bodyq** [comment]
Also @gianalarcon , I asked you a question on the discord channel

File: dojo-starter/.tool-versions [hunk 1]
```diff
@@ -1 +1 @@
-dojo 1.0.2
```
Comment (gianalarcon) [inline dojo-starter/.tool-versions:1]: we should not change this file, it is just read only
Can check [here](https://github.com/The-Marquis-Gaming/themarquis-ui/blob/1330d181b11b2332a23034e5c75c6a913c34badd/.github/workflows/contracts.yaml#L41C16-L41C24) how to read it

**Review by gianalarcon — CHANGES_REQUESTED** [review]
looks good, need to fix the comment I left, also need to attach the github action results you got on your own repo

**Comment by No-bodyq** [comment]
All done @gianalarcon

**Comment by gianalarcon** [comment]
Great job! Thanks for your contribution!

**Comment by No-bodyq** [comment]
Thank you ser

**Pull request merged**